### PR TITLE
fix(sql) : Fix sql errors on module installation

### PR DESCRIPTION
**Pull request by AssanDiomande**
## Description

**Fixes** MON-7056

Update script /bin/dsmd.pl to manage errors during installation of centeon-dsm-client and centeon-dsm-client

## Type of change

- [x] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 20.10.x
- [ ] 21.04.x
- [x] 21.10.x
- [x] 22.04.x (master)

<h2> How this pull request can be tested ? </h2>

* run this command on your terminal : yum install centreon-dsm-server centreon-dsm-client
* run this command : /usr/share/centreon/bin/dsmd.pl
* check if some errors are displayed

### CODE REVIEW

File: bin/dsmd.pl [hunk 1]
```diff
@@ -205,18 +205,21 @@ sub load_slot_locks {
     $self->{cache_locks} = {};
     my $rows = [];
     while (1) {
-        my ($status, $sth) = $self->{db_centstorage}->query("SELECT `lock_id`, `host_id`, `service_id`, `internal_id`, `id`, `ctime`, `status` FROM mod_dsm_locks");
-        if ($status != -1) {
-            while (my $row = ( shift(@$rows) || # get row from cache, or reload cache:
-                       shift(@{$rows = $sth->fetchall_arrayref(undef, $self->{dsmd_config}->{sql_fetch})||[]})) ) {
-                $self->{cache_locks}->{$$row[1] . '.' . $$row[2]} = [$$row[0], $$row[3], $$row[4], $$row[5], $$row[6]];
+        my $moduleInfoDsm = $self->{db_centstorage}->query("select count(*) from centreon.modules_informations");
```
Comment (kduret) [inline bin/dsmd.pl:208]: you should select name of module where name = centreon-dsm
```suggestion
        my $moduleInfoDsm = $self->{db_centreon}->query("select count(*) from modules_informations");
```

Comment (AssanDiomande) [reply bin/dsmd.pl:208]: I got the error : Can't call method "query" on an undefined value at /usr/share/centreon/bin/dsmd.pl line 208

File: bin/dsmd.pl [hunk 2]
```diff
@@ -547,14 +547,19 @@ sub run {
     );
     $self->{db_centstorage}->connect();
 
-    $self->load_slot_locks();
+    my $moduleDsm = $self->{db_centstorage}->query("select count(name) from centreon.modules_informations where name = 'centreon-dsm'");
```
Comment (kduret) [inline bin/dsmd.pl:550]: You can't hardcoded `centreon` database
Some users will use databases with other names

File: bin/dsmd.pl [hunk 3]
```diff
@@ -547,14 +547,19 @@ sub run {
...
+    if ($moduleDsm->fetchrow() != 0) {
+        $self->load_slot_locks();
+    }
     while (1) {
         $self->check_signals();
         $self->clean_locks();
-        if ($self->get_alarms() == 0) {
-            $self->manage_alarms();
+        my $moduleDsmCacheExist = $self->{db_centstorage}->query("select count(*) FROM information_schema.TABLES WHERE TABLE_SCHEMA = 'centreon_storage' AND TABLE_NAME = 'mod_dsm_cache'");
```
Comment (kduret) [inline bin/dsmd.pl:557]: You can't hardcoded `centreon_storage` database name
Some users will use databases with other names

Comment (kduret) [reply bin/dsmd.pl:557]: furthermore, you are doing this query in a `while (1)`, this is a performance regression ...

**Review by kduret — CHANGES_REQUESTED** [review]

File: bin/dsmd.pl (hunk 3 above)
Comment (AssanDiomande) [reply bin/dsmd.pl:557]: I'll delete this modification, the table 'mod_dsm_cache' is created after installing module from website

File: bin/dsmd.pl (hunk 2 above)
Comment (AssanDiomande) [reply bin/dsmd.pl]: variable created

File: bin/dsmd.pl [hunk 4]
```diff
@@ -532,6 +532,7 @@ sub manage_alarms {
 
 sub run {
     my ($self, %options) = @_;
+    my $dbCentreon = 'centreon';
```
Comment (kduret) [inline bin/dsmd.pl:535]: create a variable does not fix anything, you should get this value from config, like it's done for centstorage connection

**Review by kduret — CHANGES_REQUESTED** [review]

File: bin/dsmd.pl (hunk 4 above)
Comment (kduret) [reply bin/dsmd.pl:535]: something like 
```
    $self->{db_centreon} = centreon::common::db->new(
        db => $self->{centreon_config}->{centreon_db},
        host => $self->{centreon_config}->{db_host},
        port => $self->{centreon_config}->{db_port},
        user => $self->{centreon_config}->{db_user},
        password => $self->{centreon_config}->{db_passwd},
        force => 0,
        logger => $self->{logger}
    );
    $self->{db_centreon}->connect();
```

Comment (AssanDiomande) [reply bin/dsmd.pl]: done

**Review by kduret — APPROVED** [review]

File: bin/dsmd.pl (hunk 2 above)
Comment (AssanDiomande) [reply bin/dsmd.pl]: variable deleted, used configuration instead

**Pull request merged**